Footer V1

CooperW824 opened the pull request:
# Footer V1

## Please briefly describe the changes made in your pull request below

Added the footer component with all the social media links

## If your changes made visual updates to the website, please add screenshot(s) below
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/77216025/543e20c8-4d7b-4c9d-befe-39a231164acd)
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/77216025/2cd56408-3fe6-45de-a56a-160b55eafe4e)
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/77216025/f7cbb471-20d4-4f5a-a641-6279e67a456e)
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/77216025/869296de-0ccd-4e5a-8420-7b22be13da3e)

grt812 reviewed (approved):
There are a few things that I think could be improved:
1. The footer in the below image overflows with the navigation because of position: absolute on the footer
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/31299429/e96b6750-fbfd-4ba9-944f-b8ab9bddbe28)
. We can probably remove this, and it would still work as intended.
2. The icons become increasingly small due on small phone sizes because of the fixed padding  
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/31299429/0234675b-58ae-4129-900b-299a4e6b4116)
This can be fixed by replacing the m-4 class in the card with a grid space class in the flex container that has each of the cards.
3. The aesthetics of the register now button and headings combined with the font choice don't look that good IMO but that is pretty subjective.
4. I think that the design when on a larger desktop screen could be improved so that all content is centered. The "return to top" button could also be it's own position: fixed button that is to the bottom right of the screen instead of part of the footer.
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/31299429/6fc042c1-ffd6-40de-911b-90b1aca93dc8)
  
 I will make another branch where I will fix some of these problems after this is merged in and I might find some other issues that could be fixed in the process. Other than that, it's pretty solid. The icons look pretty cool.

CooperW824 commented:
I fixed suggestion 1 and 2, as for number 3, i just copied the design from the figma that marketing created at the latest meeting, we can talk with them about colors and design at the next meeting. As for suggestion 4, I just removed the return to top button as in the latest splash screen, the nav bar will always be visible if you see the footer so idk if a return to top button is needed in retrospect. I just flex-col'd everything.

grt812 commented:
> I fixed suggestion 1 and 2, as for number 3, i just copied the design from the figma that marketing created at the latest meeting, we can talk with them about colors and design at the next meeting. As for suggestion 4, I just removed the return to top button as in the latest splash screen, the nav bar will always be visible if you see the footer so idk if a return to top button is needed in retrospect. I just flex-col'd everything.

Ok cool if that’s the case then I wont make another one
